### ISSUE TITLE
Larger example?

### ISSUE THREAD

**Issue by grosch-intl** [PathogenPlayground/DiagnosticSuppressorTest#1]
Any chance of buliding this example out a bit?  I'm wanting to suppress IDE0270 if the "other" thing is throwing an exception, but I really have no idea what I'm doing with the syntax trees.

**Comment by PathogenDavid** [comment]
Hello! Sorry for the delayed response, I have not had time to pay attention to GitHub lately.

This repo is something I wrote back when `DiagnosticSuppressor` was first introduced in Roslyn. It's intentionally very bare-bones and more or less assumes you have some familiarity with writing Roslyn analyzers.

(Diagnostic suppressors are a type of analyzer. So anything you read about analyzers will apply to them too.)

Even if I had time right now I'd hesitate to complicate it any further.

> I really have no idea what I'm doing with the syntax trees.

Two tools I really like for exploring syntax trees:

* [SharpLab](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLAChcoGYACdIgYVwG9cjaTiUAWIgWQAoBKGu6nO/ogHpBRAMpgAnkQAWAUwA28gPYBCbgJIYAnGwBEACQXLoRAO5KATvNgrdHANzqiAX1zOgA===) has a mode for showing the syntax tree of a piece of code.
  * Best for small self-contained snippets you want to inspect.
* The [syntax visualizer window](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/syntax-visualizer) in Visual Studio.
  * Best for when you need to inspect code that's part of a larger project.
  * TL;DR: Install `.NET Compiler Platform SDK` with Visual Studio Installer and go to View > Other Windows > Syntax Visualizer and click somewhere in a C# file.

Another tool which is sometimes useful is [Roslyn Quoter](https://roslynquoter.azurewebsites.net/). You feed in C# code and it spits out [`SyntaxFactory`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntaxfactory) calls needed to build an equivalent syntax tree. (I don't use this one much anymore, but it's worth mentioning.)

--------

You'll likely also find it helpful to look at the implementation of IDE0270. It looks like it's spread across [`AbstractUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer`](https://github.com/dotnet/roslyn/blob/43913e6f7e163dfc64700aa828af23f89aed4f40/src/Analyzers/Core/Analyzers/UseCoalesceExpression/AbstractUseCoalesceExpressionForIfNullCheckDiagnosticAnalyzer.cs) and [`CSharpUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer`](https://github.com/dotnet/roslyn/blob/43913e6f7e163dfc64700aa828af23f89aed4f40/src/Analyzers/CSharp/Analyzers/UseCoalesceExpression/CSharpUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer.cs).

(The separation exists because [this diagnostic also applies to Visual Basic](https://github.com/dotnet/roslyn/blob/43913e6f7e163dfc64700aa828af23f89aed4f40/src/Analyzers/VisualBasic/Analyzers/UseCoalesceExpression/VisualBasicUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer.vb).)

You might actually find it easier to fork the analyzer and make your own rather than read the tea leaves to suppress the ones you don't want. (IE: Just make your own diagnostic and disable IDE0270.) I did not look closely but it might be as simple as tweaking the various `IsThrowStatement` checks. (If you go that route, you might also want to copy [the code fix](https://github.com/dotnet/roslyn/blob/43913e6f7e163dfc64700aa828af23f89aed4f40/src/Analyzers/Core/CodeFixes/UseCoalesceExpression/UseCoalesceExpressionForIfNullStatementCheckCodeFixProvider.cs) if they're important to your team's workflow.)

**Comment by grosch-intl** [comment]
Thank you so much for this amazing reply, it's incredibly helpful!
